girder_worker/tasks.py

```diff
@@ -1,25 +1,20 @@
 import core
-from .core.utils import JobManager, JobStatus
+from girder_worker.utils import JobStatus
 from .app import app
 
 
 def _cleanup(*args, **kwargs):
     core.events.trigger('cleanup')
 
 
-@app.task(name='girder_worker.run', after_return=_cleanup)
-def run(*pargs, **kwargs):
-    jobInfo = kwargs.pop('jobInfo', {})
-    retval = 0
+@app.task(name='girder_worker.run', bind=True, after_return=_cleanup)
+def run(task, *pargs, **kwargs):
+    kwargs['_job_manager'] = task.job_manager \
+        if hasattr(task, 'job_manager') else None
 
-    with JobManager(logPrint=jobInfo.get('logPrint', True),
```

Comment:
I've replaced it with celery's built-in mechanisms. Task entry is handled in the [gw_task_prerun](https://github.com/girder/girder_worker/blob/gw-worker-signals/girder_worker/app.py#L14-L41) function, and task exit is handled in [gw_task_postrun](https://github.com/girder/girder_worker/blob/gw-worker-signals/girder_worker/app.py#L71-L78).